### PR TITLE
[Test] Large scale threaded actor workload

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This adds a large scale threaded actor test. The test is as follow;

- Create many threaded actors
- Each actor keeps running computation loop while it provides result & metadata through a separate methods.
- Driver keeps getting result.
- Every X second, it kills all actors and restarts.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by scv119 — APPROVED** [review]

File: release/nightly_tests/stress_tests/test_threaded_actors.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        return result
+
+
+def start_actors(num_actors):
+    """Create actors and run the computation loop.
+    """
+    num_actors = int(num_actors)
+    start = time.time()
+    pi_actors = [
+        PiCalculator.options(max_concurrency=10).remote()
```
Comment (hauntsaninja) [inline release/nightly_tests/stress_tests/test_threaded_actors.py:78]: We place actors onto nodes ourselves, so could replace this with something like:
```
nodes = []
while len(nodes) < num_nodes:
    nodes = [
        next((r for r in n["Resources"] if "node" in r), None) for n in ray.nodes() if n["Alive"]
    ]
    nodes = [n for n in nodes if n is not None]
pi_actors = [PiCalculator.options(resources={n: 0.01}, max_concurrency=10).remote() for n in nodes for _ in range(num_cpus_per_node)]
```

Comment (hauntsaninja) [reply release/nightly_tests/stress_tests/test_threaded_actors.py:78]: We also pass some args to actor, so you could e.g. also add `PiCalculator.remote(metadata={"meta": 1})` and then in `PiCalculator` have `self.metadata = metadata`.

**Review by hauntsaninja — COMMENTED** [review]
Awesome! Looks about right, had one comment about scheduling

File: release/nightly_tests/stress_tests/test_threaded_actors.py (hunk 1 above)
Comment (rkooo567) [reply release/nightly_tests/stress_tests/test_threaded_actors.py:78]: Thanks for this 2! Will definitely add them.

File: release/nightly_tests/stress_tests/test_threaded_actors.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+    - It first creates actors as many as num_cpus with max_concurrency=10
+    - Each actor computes pi and put the result to the queue.
+    - Driver keeps getting result & metadata from the actor.
+    - Every X seconds, it kills all actors and restarts them.
+    """
+    ray.init(address="auto")
+    args, unknown = parse_script_args()
+    num_cpus = ray.cluster_resources()["CPU"]
+    num_nodes = len([1 for n in ray.nodes() if n["Alive"]])
```
Comment (hauntsaninja) [inline release/nightly_tests/stress_tests/test_threaded_actors.py:121]: nit:
```suggestion
    num_nodes = sum(1 for n in ray.nodes() if n["Alive"])
```

File: release/nightly_tests/stress_tests/test_threaded_actors.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            next((r for r in n["Resources"] if "node" in r), None)
+            for n in ray.nodes() if n["Alive"]
+        ]
+        nodes = [n for n in nodes if n is not None]
+    pi_actors = [
+        PiCalculator.options(resources={
+            n: 0.01
+        }, max_concurrency=10).remote({
+            "meta": 1
+        }) for n in nodes for _ in range(num_actors)
```
Comment (hauntsaninja) [inline release/nightly_tests/stress_tests/test_threaded_actors.py:91]: I think this starts `num_actors * num_nodes` actors, where num_actors is total number of cpus in cluster?

Comment (rkooo567) [reply release/nightly_tests/stress_tests/test_threaded_actors.py:91]: here num_actors == num_cpus per node * num_nodes! So you are right

**Comment by rkooo567** [comment]
@hauntsaninja do you typically have 1 loop, or multiple of them? I am thinking to actually run 2~3 run RPCs, so it runs multiple of while loop concurrently

**Comment by hauntsaninja** [comment]
We only really have the one loop

**Comment by rkooo567** [comment]
This PR only adds tests, so any failure should be unrelated

**Pull request merged**